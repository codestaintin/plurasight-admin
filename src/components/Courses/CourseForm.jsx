import React from 'react';
import PropTypes from 'prop-types';
import SelectInput from '../Commons/SelectInput';
import TextInput from '../Commons/TextInput';

const CourseForm = ({
  course, allAuthors, onChange, saving, errors, onSubmit
}) => (
  <form onSubmit={onSubmit}>
    <h1>Manage Courses</h1>
    <TextInput
      name="title"
      label="Title"
      placeholder="Course title"
      value={course.title}
      onChange={onChange}
      error={errors.title}
    />

    <SelectInput
      name="authorId"
      label="Author"
      value={course.authorId}
      defaultOption="Select Author"
      options={allAuthors}
      onChange={onChange}
      error={errors.authorId}
    />

    <TextInput
      name="category"
      label="Category"
      placeholder="Course category"
      value={course.category}
      onChange={onChange}
      error={errors.category}
    />

    <TextInput
      name="length"
      label="Length"
      placeholder="Course Length e.g 1:10"
      pattern="^([0-1]?[0-9]|2[0-4]):([0-5][0-9])(:[0-5][0-9])?$"
      value={course.length}
      onChange={onChange}
      error={errors.length}
    />

    <input
      type="submit"
      disabled={saving}
      value={saving ? 'Saving...' : 'Save'}
      className="btn btn-primary"
    />
  </form>
);

CourseForm.propTypes = {
  course: PropTypes.object.isRequired,
  allAuthors: PropTypes.array.isRequired,
  onChange: PropTypes.func.isRequired,
  saving: PropTypes.bool,
  errors: PropTypes.object,
  onSubmit: PropTypes.func.isRequired
};

export default CourseForm;
