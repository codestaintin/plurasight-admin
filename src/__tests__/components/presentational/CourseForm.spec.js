import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow } from 'enzyme';
import CourseForm from '../../../components/Courses/CourseForm';

configure({ adapter: new Adapter() });

describe('<CourseForm>', () => {
  const props = {
    course: {},
    allAuthors: [],
    onChange: jest.fn(() => {}),
    onSubmit: jest.fn(() => {}),
    errors: {
      title: ''
    }
  };

  it('renders the <CourseForm> component', () => {
    const tree = shallow(<CourseForm {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
