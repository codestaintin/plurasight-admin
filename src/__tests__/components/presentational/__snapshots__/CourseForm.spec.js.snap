// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CourseForm> renders the <CourseForm> component 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <h1>
    Manage Courses
  </h1>
  <TextInput
    error=""
    label="Title"
    name="title"
    onChange={[MockFunction]}
    placeholder="Course title"
  />
  <SelectInput
    defaultOption="Select Author"
    label="Author"
    name="authorId"
    onChange={[MockFunction]}
    options={Array []}
  />
  <TextInput
    label="Category"
    name="category"
    onChange={[MockFunction]}
    placeholder="Course category"
  />
  <TextInput
    label="Length"
    name="length"
    onChange={[MockFunction]}
    pattern="^([0-1]?[0-9]|2[0-4]):([0-5][0-9])(:[0-5][0-9])?$"
    placeholder="Course Length e.g 1:10"
  />
  <input
    className="btn btn-primary"
    type="submit"
    value="Save"
  />
</form>
`;
